### ISSUE TITLE
Doesn't revalidate invalid cells

### ISSUE THREAD

**Issue by Oleniacz** [specify/specify7#227]
The original invalid cells were entered as 0000-01-01, 1990-15-10 (which received a note saying that they were unparseable dates) and 1990/10/00 (which received a note saying that it was an invalid date). 

1) When I changed the date to test leap days, I entered 2/29/2012 and 2/29/2015 expecting the latter to come back invalid. However, they both came back invalid and when my cursor roamed over the cell the bubble with the note said that it was still the same unparseable dates of 00-01-01 and 1990-15-10 respectively. 

![unparseable1](https://cloud.githubusercontent.com/assets/17576913/13438570/278114d6-dfaf-11e5-9ead-4f0422af8dd4.png)

![unparseable2](https://cloud.githubusercontent.com/assets/17576913/13438779/1db00ec0-dfb0-11e5-8774-e4629c85522e.png)

2) In addition, I entered yesterday's date 2/29/2016 in the same cell that previously had 1990/10/00 and came back saying it was an invalid date. I saved and validated and the cell still said that it is an invalid date even though it is a valid date. I wrote it in the cell below which wasn't an invalid cell, saved and validated and it was fine. 

![invaliddatewhenitsnot](https://cloud.githubusercontent.com/assets/17576913/13438572/2782f6e8-dfaf-11e5-8727-77bbbc110bd4.png)

In summary, I found that invalid cells were sticking with their original notes and errors and not revalidated and updating the current invalid issue, and were not updating after date was properly entered.

**Comment by garrettgirard** [comment]
Verified in KU Birds. Cells that are at first invalid, then fixed and revalidated no longer show validation error messages from the initial validation unless the cell is still invalid.
